sw/cheri/boot/boot_loader.cc

```diff
@@ -62,6 +92,7 @@ static void write_hex_with_prefix(UartRef uart, const char *msg, uint32_t value)
 }
 
 uint32_t read_elf(SpiFlash                  &flash,
+				  SoftwareSlotAddress        addr,
```

Comment:
I wonder rather see this as a uint32_t directly to avoid the casts inside this function. Cast can happen before calling `read_elf`.